components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -678,9 +722,9 @@ nsapi_error_t ESP8266::send(int id, const void *data, uint32_t amount)
         tr_debug("send(): Connection disrupted.");
     }
 
-    if (!_sock_i[id].open && ret != NSAPI_ERROR_OK) {
+    if (!_sock_i[id].open && ret < NSAPI_ERROR_OK) {
```

Comment:
Smaller than zero would be better because we know that negative values are error codes.